### ISSUE TITLE
About s3dis  preprocessing 

### ISSUE THREAD

**Issue by jing-zhao9** [Pointcept/Pointcept#207]
Thank you very much for your work！
I have a question: How is the S3DIS data set preprocessed into PTv3 input format? Is it something like PointNet++ where you cut it into a grid and you input 4096 points into each grid。

**Comment by Gofinge** [comment]
We don't crop real-world point clouds during preprocessing. Batched point cloud in pointcept no need to be even.
https://github.com/Pointcept/Pointcept?tab=readme-ov-file#offset

**Comment by aldinorizaldy** [comment]
This is important as Pointnet's approach by cutting pointclouds into smaller blocks limit the networks to see beyond the size of the blocks. In many airborne point cloud data, having larger block size is important to capture large objects such as warehouse. However, it is problematic since this method resamples the number of points into a fix number e.g. 4096 points. Larger block with less points means the density would be lower. 

As you say Pointcept does not crop point clouds. I also read from here: https://github.com/Pointcept/Pointcept/issues/91 if we have very large point cloud then it is suggested to cut the data into smaller chunks. So, I would assume Pointcept can 'see' the whole area of each point cloud which is very good compared to Pointnet's approach that is limited to the size of the block. 

I checked this https://github.com/Pointcept/Pointcept?tab=readme-ov-file#offset, but I still don't really understand. My question is:
1. If we have batch size = 12 in config, does that mean we have 12 real-world point clouds (without cropping) in 1 batch?
2. What makes me confused is compared to Pointnet/Pointnet++, they usually have similar batch size but the number of points of each block (cropped point cloud) is small (e.g., 4096 points). Could you please explain a bit more about the concept of offset here? And why we can use the uncropped point clouds for the input of Pointcept without having out-of-memory?

Thanks!!

**Comment by Gofinge** [comment]
> If we have batch size = 12 in config, does that mean we have 12 real-world point clouds (without cropping) in 1 batch?

Yes, 12  real-world point clouds in one single batch. Yet typically, we set num GPUs as 4, which means 3 point clouds on a single GPU.

But sometimes, we do crop point cloud as data augmentation (SphereCrop, for indoor data), as we can not feed the whole point cloud to the network once. But as augmentation, the random crop is much softer than fixed cropping during preprocessing, resulting in a better performance.

>  They usually have similar batch size but the number of points of each block (cropped point cloud) is small (e.g., 4096 points).

These classical backbones are not efficient. As I said in our paper, accelerating operators that KNN or FPS is non-trivial.

> Could you please explain a bit more about the concept of offset here?

So, each point cloud in a single batch has a different number of points. We need to introduce something to record the separator of each point cloud in the batch.

> And why we can use the uncropped point clouds for the input of Pointcept without having out-of-memory?

As mentioned above, we do have some random crops as data augmentation. Also, backbones like PTv3, SpUNet are really efficient, which mean they naturally can handle a large number of points.
